qa/SidechainTestFramework/sidechainauthproxy.py

```diff
@@ -71,7 +71,7 @@ def __getattr__(self, name):
             name = "%s.%s" % (self.__service_name, name)
         return SidechainAuthServiceProxy(self.__service_url, name, connection=self.__conn)
 
-    def _request(self, method, path, postdata):
+    def _request(self, method, path, postdata, additional_header):
```

Comment:
I would suggest to update the constructor of the class with "api key" argument. The value is known at the place of proxy instance creation.

Then, for every call to the API automatically add `apiKey` to the header.
So, inside the particular test we don't need to care to set:
```
auth = {
         "api_key": api_key
      }
```
For the authorization checks we may keep this argument `additional_header` but with a different context - "customApiKey".
By default , method looks like `sidechainNode.wallet_exportSecret(request)`, but if you want to specify custom api key it will look like `sidechainNode.wallet_exportSecret(request, customApiKey)`.

Reply:
Done